### PR TITLE
fix a bunch of issues with pending txns

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1519

Our transactions solely used balance change information to display the transaction rows / details sheet. For pending transactions those balances haven't updated / changed yet, so we need to rely on the quote information while the transaction is in a `pending` status.

**Note this is a fix for both swaps v1 and v2**

- [x] test other transaction types and make sure no regressions have been made

### CODE REVIEW

File: src/screens/transaction-details/components/TransactionMasthead.tsx [hunk 1]
```diff
@@ -289,8 +314,28 @@ export default function TransactionMasthead({ transaction }: { transaction: Rain
...
   const outputAsset = useMemo(() => {
-    const outAsset = transaction?.changes?.find(a => a?.direction === 'out')?.asset;
-    if (!outAsset) return undefined;
+    const change = transaction?.changes?.find(a => a?.direction === 'out');
+
+    if (!change?.asset) return undefined;
+
+    // NOTE: For pending transactions let's use the change value
+    // since the balance hasn't been updated yet.
+    if (['swap', 'wrap', 'unwrap'].includes(transaction.type) && transaction.status === 'pending') {
```
Comment (benisgold) [inline src/screens/transaction-details/components/TransactionMasthead.tsx:323]: i've seen this condition a few times, could be good to extract it out into a helper called isPendingSwap or something

File: src/raps/actions/swap.ts [hunk 2]
```diff
@@ -323,25 +342,34 @@ export const swap = async ({
...
     nonce: swap.nonce,
     status: 'pending',
     type: 'swap',
+    swap: {
+      type: SwapType.normal,
+      fromChainId: parameters.assetToSell.chainId as unknown as SwapChainId,
+      toChainId: parameters.assetToBuy.chainId as unknown as SwapChainId,
+
+      // TODO: Is this right?
+      isBridge:
```
Comment (benisgold) [inline src/raps/actions/swap.ts:365]: what is this used for? it's also possible to execute a bridge tx with two different tokens, but not sure if that is relevant here

Comment (walmat) [reply src/raps/actions/swap.ts:365]: I don't see it being used anywhere currently so not sure if it's even necessary. I can probably just remove it.

**Review by greg-schrammel — APPROVED** [review]

**Pull request merged**